src/utils/pyproject.ts

```diff
@@ -40,9 +47,21 @@ function parsePyproject(pyprojectContent: string): string | undefined {
   )
     .flat()
     .filter((item: string | object) => typeof item === "string");
-  return getRuffVersionFromAllDependencies(
+  const version = getRuffVersionFromAllDependencies(
     dependencies.concat(optionalDependencies, devDependencies),
   );
+  if (version) return version;
```

Comment:
```suggestion
  const allDependencies = dependencies.concat(optionalDependencies, devDependencies);
  return getRuffVersionFromAllDependencies(allDependencies) || getRuffVersionFromPoetryGroups(pyproject);
```

Lets put the poetry handling in a dedicated function `getRuffVersionFromPoetryGroups`.

Otherwise looks really good!